### DISCUSSION TITLE
Any way to make AlsMovementComponent::SetMovementModeLocked() accessible to components in derived classes?

### DISCUSSION THREAD

**Discussion by colorindarkness** [Sixze/ALS-Refactored#235] [General]
I'm creating a new component to extend ALSXT, and I've been using EnableInput()/DisableInput() but this will not work on AI. So is there any way to make SetMovementModeLocked() accessible? Especially to components? I would like to replicate the methods used in Mantling/Als_Actions, but I want it to work with AI as well. 
I've tried creating a Setter function in my derived class but that does not work either.
Any ideas?

**Comment by Sixze** [comment]
Isn't this function public? You can access it anywhere where you have access to the character movement component.

**Reply by colorindarkness** [reply]
Yes it's a public function, but the AlsMovmementComponent is a protected component, does that change anything? I made the following public function in my Derived Character class:
```
void AALSXTCharacter::SetMovementModeLocked(bool bNewMovementModeLocked)
{
	AlsCharacterMovement->SetMovementModeLocked(bNewMovementModeLocked);
}
```
But when I call it from the Actor Component (UnarmedCombatComponent) it does not seem to work. Is there anything I'm doing wrong? Or is it impossible to do from a component or derived class?

**Reply by ameaninglessname** [reply]
>  but the AlsMovmementComponent is a protected component

You can do this : `Character->GetCharacterMovement<UAlsCharacterMovementComponent>()`

**Reply by colorindarkness** [reply]
Ok! I'm trying that but it gives a "Type Name is Not Allowed" error. (`error C2275: 'UAlsCharacterMovementComponent': illegal use of this type as an expression`). I made sure the header was included as well

Trying the following creates no errors, but seems to do nothing as the Character can still move
`Cast<UAlsCharacterMovementComponent>(GetCharacterMovement())->SetMovementModeLocked(bNewMovementModeLocked);`

**Comment by Sixze** [comment]
If you want to block input for both the player and AI, it's better to just override `UPawnMovementComponent::ConsumeInputVector()`, call `Super::::ConsumeInputVector()` from the override, ignore the value returned by that call, and return `FVector::ZeroVector` instead.

This is how I blocked input in the projects I worked on.
